lib/resque_message_handler.rb

```diff
@@ -125,19 +125,24 @@ def update
   private
 
   # Given a hash relating barcodes to scsb items, returns a new hash consisting
-  # of only those items that have the named status (`:available` or
-  # `:unavailable`). To ensure a clean segmentation, we only match `:available`
-  # to `"availality": "Available"`
-  def select_by_status(barcode_mapping, status)
+  # of only those items that are either:
+  #  1. Available
+  #  2. Incomplete
+  # These are the classes of records we're able to update.
+  def get_barcodes_allowing_updates(barcode_mapping)
     barcode_mapping.select do |barcode, item|
-      if status == :available
-        item['availability'] == 'Available'
-      else
-        item['availability'] != 'Available'
-      end
+      item['availability'] == 'Available' || item['title'] == 'Dummy Title'
```

Comment:
So is the issue just that we were not looking at all the relevant fields to determine availability? What exactly is the significance of these checks?